naucse/templates/lesson.html

```diff
@@ -22,17 +22,17 @@
 
         <hr class="lesson-end">
 
-        {% if (prv is defined and prv) or (nxt is defined and nxt) %}
+        {% if (prv_link is defined and prv_title is defined) or (nxt_link is defined and nxt_title is defined) %}
         <div class="row prev-next">
-            {% if prv %}
+            {% if prv_link %}
                 <div class="col text-left">
-                    <a href="{{ lesson_url(prv.lesson, page=prv.slug) }}">← {{ prv.title }}</a>
+                    <a href="{{ lesson_url(('/').join(prv_link.split('/')[:2])) }}">← {{ prv_title }}</a>
                 </div>
             {% endif %}
 
-            {% if nxt %}
+            {% if nxt_link %}
                 <div class="col text-right">
-                    <a href="{{ lesson_url(nxt.lesson, page=nxt.slug) }}">{{ nxt.title }} →</a>
+                    <a href="{{ lesson_url(('/').join(nxt_link.split('/')[:2])) }}">{{ nxt_title }} →</a>
```

Comment:
URL manipulation should be left to the URL converters and `@app.route()`. Avoid having logic involving `/` elsewhere – you're making too many assumptions about the structure of the object's `str()` representation.